src/framebounds.c

```diff
@@ -21,6 +21,26 @@ struct plum_rectangle * get_frame_boundaries (struct context * context, bool anc
 
 void adjust_frame_boundaries (const struct plum_image * image, struct plum_rectangle * restrict boundaries) {
   uint64_t empty_color = get_empty_color(image);
+  #define checkframe(type) do                                                                                                                             \
```

Comment:
Here the palette code was so similar to the 16/32/64 color code that I wanted to use the existing `checkframe` macro for the palette as well.